Lake/Load/Main.lean

```diff
@@ -23,78 +23,93 @@ def loadDepsFromEnv (env : Environment) (opts : Options) : Except String (Array
   packageDepAttr.ext.getState env |>.foldM (init := #[]) fun arr name => do
     return arr.push <| ← evalConstCheck env opts Dependency ``Dependency name
 
-/--
-Resolve a single dependency and load the resulting package.
-Resolution is based on the `Dependency` configuration and the package manifest.
--/
-def resolveDep (packagesDir : FilePath) (pkg : Package) (dep : Dependency)
-(topLevel : Bool) (shouldUpdate : Bool) : StateT Manifest LogIO Package := do
-  let entry? := (← get).find? dep.name
-  let entry? := entry?.map fun entry =>
-    {entry with shouldUpdate := if topLevel then shouldUpdate else entry.shouldUpdate}
-  let ⟨dir, url?, tag?, entry?⟩ ← materializeDep packagesDir pkg.dir dep entry?
-  let configEnv ← elabConfigFile dir dep.options pkg.leanOpts (dir / defaultConfigFile)
-  let config ← IO.ofExcept <| PackageConfig.loadFromEnv configEnv pkg.leanOpts
-  let depPkg : Package := {
-    dir, config, configEnv,
-    leanOpts := pkg.leanOpts
-    remoteUrl? := url?, gitTag? := tag?
-  }
-  unless depPkg.name = dep.name do
-    error <|
-      s!"{pkg.name} (in {pkg.dir}) depends on {dep.name}, " ++
-      s!"but resolved dependency has name {depPkg.name} (in {dir})"
-  if let some entry := entry? then
-    modify (·.insert entry)
-  let depManifest ← Manifest.loadOrEmpty depPkg.manifestFile
-  for depEntry in depManifest do
-    if let some entry := (← get).find? depEntry.name then
-      let shouldUpdate :=
-        match entry.inputRev?, depEntry.inputRev? with
-        | none,     none     => entry.rev != depEntry.rev
-        | none,     some _   => false
-        | some _,   none     => false
-        | some rev, some dep => rev = dep ∧ entry.rev ≠ depEntry.rev
-      let contributors := entry.contributors.insert depPkg.name depEntry.toPersistentPackageEntry
-      modify (·.insert {entry with contributors, shouldUpdate})
-    else
-      let contributors := NameMap.empty.insert depPkg.name depEntry.toPersistentPackageEntry
-      modify (·.insert {depEntry with contributors})
-  return depPkg
+def loadDepPackage (parentPkg : Package) (result : MaterializeResult)
+(dep : Dependency) : LogIO Package := do
+  let dir := result.pkgDir
+  let configEnv ← elabConfigFile dir dep.options parentPkg.leanOpts (dir / defaultConfigFile)
+  let config ← IO.ofExcept <| PackageConfig.loadFromEnv configEnv parentPkg.leanOpts
+  return { dir, config, configEnv, leanOpts := parentPkg.leanOpts }
```

Comment:
```suggestion
  return { dir, config, configEnv, leanOpts := parentPkg.leanOpts, remoteUri? := result.remoteUri?, gitTag? := result.gitTag? }
```

The proper setting of the uri/tag was dropped somewhere during this PR.

Reply:
Ah the bane of default field values...